src/chartsFactory.jsx

```diff
@@ -1,79 +1,90 @@
-        ...config,
-        chart: {
-          ...chartConfig,
-          renderTo: this.chartRef
+function chartsFactory(chartType, Highcharts) {
+
+    class Charts extends Component {
+        constructor() {
+            super()
+            this.chartType = chartType;
+            this.Highcharts = Highcharts;
+            this.displayName = 'Highcharts' + chartType;
+        }
+
+        setChartRef(chartRef) {
+            this.chartRef = chartRef;
         }
-      }, this.props.callback);
 
-      if (!this.props.neverReflow) {
-        win && win.requestAnimationFrame && requestAnimationFrame(()=>{
-          this.chart && this.chart.options && this.chart.reflow();
-        });
-      }
-    },
+        renderChart(config) {
+            if (!config) {
+                throw new Error('Config must be specified for the ' + this.displayName + ' component');
+            }
+            const chartConfig = config.chart;
+            this.chart = new this.Highcharts[this.chartType]({
+                ...config,
+                chart: {
+                    ...chartConfig,
+                    renderTo: this.chartRef
+                }
+            }, this.props.callback);
 
-    shouldComponentUpdate(nextProps) {
-      if (nextProps.neverReflow || (nextProps.isPureConfig && this.props.config === nextProps.config)) {
-        return true;
-      }
-      this.renderChart(nextProps.config);
-      return false;
-    },
+            if (!this.props.neverReflow) {
+                win && win.requestAnimationFrame && requestAnimationFrame(() => {
+                    this.chart && this.chart.options && this.chart.reflow();
+                });
+            }
+        }
 
-    getChart: function (){
-      if (!this.chart) {
-        throw new Error('getChart() should not be called before the component is mounted');
-      }
-      return this.chart;
-    },
+        shouldComponentUpdate(nextProps) {
+            if (nextProps.neverReflow || (nextProps.isPureConfig && this.props.config === nextProps.config)) {
+                return true;
+            }
+            this.renderChart(nextProps.config);
+            return false;
+        }
 
-    componentDidMount: function (){
-      this.renderChart(this.props.config);
-    },
+        getChart() {
+            if (!this.chart) {
+                throw new Error('getChart() should not be called before the component is mounted');
+            }
+            return this.chart;
+        }
 
-    componentWillUnmount() {
-      this.chart.destroy();
-    },
+        componentDidMount() {
+            this.renderChart(this.props.config);
+        }
 
-    render: function (){
-      return <div ref={this.setChartRef} {...this.props.domProps} />;
+        componentWillUnmount() {
+            this.chart.destroy();
+        }
+
+        render() {
+            return <div ref={this.setChartRef.bind(this)} {...this.props.domProps} />;
+        }
+    }
+
+    if (cutCode) {
```

Comment:
rename to isProdMode or something similar?

Reply:
Ok